PyInstaller/building/build_main.py

```diff
@@ -374,10 +374,28 @@ def assemble(self):
         if is_win:
             depmanifest.writeprettyxml()
 
-        # The first script in the analysis is the main user script. Its node is used as
-        # the "caller" node for all others. This gives a connected graph rather than
-        # a collection of unrelated trees, one for each of self.inputs.
-        # The list of scripts is in self.inputs, each as a normalized pathname.
+
+        ### Hook cache.
+        logger.info('Caching module hooks...')
+
+        # List of all directories containing hook scripts. Default hooks are
+        # listed before and hence take precedence over custom hooks.
+        module_hook_dirs = [get_importhooks_dir()] + self.hookspath
+
+        # Lazily defer loading hook scripts until their first access.
+        module_hook_cache = ModuleHookCache(module_graph=self.graph)
+        module_hook_cache.cache_hook_dirs(module_hook_dirs)
```

Comment:
Because: **hand-waving.**

Actually, I'll add an optional parameter to `ModuleHookCache()` supporting that. Have a Swedish lollipop. :lollipop: